funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -275,24 +260,18 @@ def start(self):
 
         while not self._kill_event.is_set():
             if self._quiesce_event.is_set():
-                log.warning("Interchange will retry connecting in 5s")
-                time.sleep(5)
+                idle_for = random.uniform(2.0, 10.0)
+                log.warning(f"Interchange will retry connecting in {idle_for:.2f}s")
+                time.sleep(idle_for)
                 self._quiesce_event.clear()
             else:
                 log.debug("Starting threads and main loop")
 
             try:
                 self._start_threads_and_main()
-            except GlobusAPIError as e:
-                # These are generated by globus_sdk/client.py::error_class
-                # for a +400 response server error, and we want to stop the
-                # retry loop on specific codes
-                if e.http_status == 409 or e.http_status == 423:
-                    # RESOURCE_CONFLICT or RESOURCE_LOCKED
-                    log.warning(f"Endpoint cannot start: {e.message}, Exiting.")
-                    self._kill_event.set()
-                else:
-                    log.exception(f"Unhandled GlobusAPIError: {e.message}.")
+            except pika.exceptions.ProbableAuthenticationError as e:
```

Comment:
I didn't know `ProbablyAuthenticationError` was a thing

Reply:
I think this was a larger question of "what's this?" that got answered verbally.  For posterity, let me recreate that explanation here.

In short, this is the exception raised when the AMQP credentials fail.

The fundamental change of this PR is to [remove the re-registration every loop iteration](https://github.com/funcx-faas/funcX/pull/1006/files#diff-7563724694028806f24db920a07eeb38565d48adec8ef004bc9a96a32e76a74bL302-L313):
```diff
          log.info("EndpointInterchange shutdown complete.")
  
      def _start_threads_and_main(self):
-         # re-register on every loop start
-         if not self.initial_registration_complete:
-             self.register_endpoint()
- 
-         self.initial_registration_complete = False
- 
          self._task_puller_proc = self.migrate_tasks_to_internal(
              self.task_q_info,
              self.endpoint_id,
```

As we still need AMQP credentials, this changes necessitates a [change in the constructor to require a _populated_ `reg_info`](https://github.com/funcx-faas/funcX/pull/1006/files#diff-7563724694028806f24db920a07eeb38565d48adec8ef004bc9a96a32e76a74bL57-L117):
```diff
      def __init__(
          self,
          config: funcx_endpoint.endpoint.utils.config.Config,
-         reg_info: t.Tuple[t.Dict, t.Dict] = None,
+         reg_info: dict[str, dict],
          logdir=".",
          endpoint_id=None,
          endpoint_dir=".",
```
```diff
              funcx_client = FuncXClient()
          self.funcx_client = funcx_client
  
-         self.initial_registration_complete = False
-         self.task_q_info, self.result_q_info = None, None
-         if reg_info:
-             self.task_q_info, self.result_q_info = reg_info
-             self.initial_registration_complete = True
+         self.task_q_info = reg_info["task_queue_info"]
+         self.result_q_info = reg_info["result_queue_info"]
  
          self.time_to_quit = False
          self.heartbeat_period = self.config.heartbeat_period
```
Upon quiesce, then, the only thing we do is use the _same credentials provided at initialization_ to try to reopen the connection to the AMQP service.  This happens at the beginning of `_main_loop`:
```python
        results_publisher = ResultQueuePublisher(
            queue_info=self.result_q_info,
        )

        with results_publisher:
```
And if those credentials are invalid (e.g., updated or AMQP user removed), then the `pika` library fails with that exception.  In the logs, that will now manifest as something like:
```
... 09:56:27 DEBUG ....endpoint.interchange:327 _main_loop _main_loop begins
... 09:56:27 WARNING ....endpoint.rabbit_mq.task_queue_subscriber:90 _on_open_failed Connection failed to open : (<SelectConnection CLOSED transport=None params=<URLParameters host=... port=5672 virtual_host=/ ssl=False>>, ProbableAuthenticationError: Client was disconnected at a connection stage indicating a probable authentication error: ("ConnectionClosedByBroker: (403) 'ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile.'",))
... 09:56:27 ERROR ....endpoint.interchange:285 start Unable to connect to AMQP service: ConnectionClosedByBroker: (403) 'ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile.'
```
Folllowed by an immediate endpoint complete shutdown (note the timestamps):
```
... 09:56:28 INFO ....endpoint.rabbit_mq.task_queue_subscriber:329 event_watcher Shutdown complete
... 09:56:28 INFO ....endpoint.interchange:223 quiesce Quiesce done
... 09:56:28 INFO ....high_throughput.executor:848 shutdown Attempting HighThroughputExecutor shutdown
... 09:56:28 INFO ....high_throughput.executor:858 shutdown Finished HighThroughputExecutor shutdown attempt
... 09:56:28 INFO ....endpoint.interchange:299 start EndpointInterchange shutdown complete.
... 09:56:28 CRITICAL ....endpoint.endpoint:349 daemon_launch Interchange terminated.
```